.github/workflows/test.yml

```diff
@@ -0,0 +1,36 @@
+name: tests # runs unit and integration tests
+
+on:
+  push:
+    branches-ignore: [main] # TODO: update this when renaming default branch to main
+  workflow_dispatch:
+
+jobs:
+  linux-tests:
+    strategy:
+      matrix:
+        node: [10, 12, 14, 16]
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node_version }}
+          cache: npm
+      - name: Cache node modules
+        id: cache-nodemodules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: "**/node_modules"
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+      - run: npm install --legacy-peer-deps
```

Comment:
why is this step required?

Reply:
The `--legacy-peer-deps` flag was added because there seems to be a peer dependency conflict due to eslint that causes `npm install` to fail. Here's an example of `npm install` failing:

https://github.com/SalesforceCommerceCloud/commerce-sdk/actions/runs/3576294833/jobs/6013930295

Reply:
are the conflicting versions are from this repo. can we resolve the conflict by upgrading the version?

Reply:
it might, let me go ahead and try changing around versioning. If it seems like this requires more work than a few minor tweaks, I'll create a follow up story